src/relations/postgresql_provider.py

```diff
@@ -286,12 +340,32 @@ def update_tls_flag(self, tls: str) -> None:
                 self.database_provides.set_tls_ca(relation.id, ca)
 
     def _update_unit_status(self, relation: Relation) -> None:
-        """# Clean up Blocked status if it's due to extensions request."""
-        if (
-            self.charm._has_blocked_status
-            and self.charm.unit.status.message == INVALID_EXTRA_USER_ROLE_BLOCKING_MESSAGE
-            and not self.check_for_invalid_extra_user_roles(relation.id)
-        ):
+        """Clean up Blocked status if it's due to extensions request."""
+        if self.charm._has_blocked_status and self.charm.unit.status.message in [
+            INVALID_EXTRA_USER_ROLE_BLOCKING_MESSAGE,
+            NO_ACCESS_TO_SECRET_MSG,
+            FORBIDDEN_USER_MSG,
+        ]:
+            if self.check_for_invalid_extra_user_roles(relation.id):
+                self.charm.unit.status = BlockedStatus(INVALID_EXTRA_USER_ROLE_BLOCKING_MESSAGE)
+                return
+            for relation in self.charm.model.relations.get(self.relation_name, []):
+                try:
+                    if secret_uri := self.database_provides.fetch_relation_field(
+                        relation.id, "requested-entity-secret"
+                    ):
+                        content = self.framework.model.get_secret(id=secret_uri).get_content()
+                        for key in content:
+                            if key in SYSTEM_USERS:
```

Comment:
I think if the user is requested in two relations, this function is going to clean up the Blocked status which is set in `_on_database_requested` event.

Reply:
Added the check for existing users when unblocking as well.